src/oidcendpoint/oidc/token.py

```diff
@@ -54,21 +102,18 @@ def _access_token(self, req, **kwargs):
                 error="invalid_request", error_description="Missing code"
             )
 
-        # Session might not exist or _access_code malformed
-        try:
-            _info = _sdb[_access_code]
-        except KeyError:
-            return self.error_cls(
-                error="invalid_grant", error_description="Code is invalid"
-            )
+        _session_info = _mngr.get_session_info_by_token(_access_code, grant=True)
+        grant = _session_info["grant"]
 
-        _authn_req = _info["authn_req"]
+        code = _mngr.find_token(_session_info["session_id"], _access_code)
+        _authn_req = grant.authorization_request
 
-        # assert that the code is valid
-        if _context.sdb.is_session_revoked(_access_code):
-            return self.error_cls(
-                error="invalid_grant", error_description="Session is revoked"
-            )
+        if "state" in req:
+            # verify that state in this request is the same as the one in the
+            # authorization request
+            if _authn_req["state"] != req["state"]:
+                logger.error("State value mismatch")
+                return self.error_cls(error="invalid_request")
```

Comment:
This must be removed, there is no `state` in the token request

Reply:
OK